src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -0,0 +1,695 @@
+                            Transaction = this.ToTransactionData(transaction, HDPayment.GetAllPayments(conn, transaction.OutputTxTime, transaction.OutputTxId, transaction.OutputIndex))
+                        });
+                    }
+                }
+
+                yield return new AccountHistory()
+                {
+                    Account = this.ToHdAccount(account),
+                    History = history
+                };
+            }
+        }
+
+        private IEnumerable<TempTable> TransactionsToLists(DBConnection conn, IEnumerable<Transaction> transactions, ChainedHeader header, uint256 fixedTxId = null, ProcessBlocksInfo processBlocksInfo = null)
+        {
+            // Convert relevant information in the block to information that can be joined to the wallet tables.
+            TempTable outputs = null;
+            TempTable prevOuts = null;
+
+            TransactionsOfInterest transactionsOfInterest = processBlocksInfo.TransactionsOfInterest;
+            AddressesOfInterest addressesOfInterest = processBlocksInfo.AddressesOfInterest;
+            int? walletId = processBlocksInfo.Wallet?.WalletId;
+
+            foreach (Transaction tx in transactions)
+            {
+                // Build temp.PrevOuts
+                uint256 txId = fixedTxId ?? tx.GetHash();
+
+                foreach (TxIn txIn in tx.Inputs)
+                {
+                    if (transactionsOfInterest?.Contains(txIn.PrevOut.Hash, conn, walletId) ?? true)
+                    {
+                        if (prevOuts == null)
+                            prevOuts = TempTable.Create<TempPrevOut>();
+
+                        // We don't know which of these are actually spending from our
+                        // wallet addresses but we record them for batched resolution.
+                        prevOuts.Add(new TempPrevOut()
+                        {
+                            OutputTxId = txIn.PrevOut.Hash.ToString(),
+                            OutputIndex = (int)txIn.PrevOut.N,
+                            SpendBlockHeight = header?.Height ?? 0,
+                            SpendBlockHash = header?.HashBlock.ToString(),
+                            SpendTxIsCoinBase = (tx.IsCoinBase || tx.IsCoinStake) ? 1 : 0,
+                            SpendTxTime = (int)tx.Time,
+                            SpendTxId = txId.ToString(),
+                            SpendTxTotalOut = tx.TotalOut.ToDecimal(MoneyUnit.BTC)
+                        });
+
+                        transactionsOfInterest?.AddTentative(txId);
+                    }
+                }
+
+                // Build temp.Outputs.
+                for (int i = 0; i < tx.Outputs.Count; i++)
+                {
+                    TxOut txOut = tx.Outputs[i];
+
+                    if (txOut.IsEmpty)
+                        continue;
+
+                    if (txOut.ScriptPubKey.ToBytes(true)[0] == (byte)OpcodeType.OP_RETURN)
```

Comment:
I believe there's some sort of `IsUnspendable` helper method that can be used instead of this

Reply:
Hmm one would think so, but I could not find it.